cpp/foxglove/include/foxglove/mcap.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "foxglove-c/foxglove-c.h"
```

Comment:
```suggestion
#include <foxglove-c/foxglove-c.h>
```

I'm curious for your opinion on this - I've been trying to avoid including this header from the C++ headers, because then it will pollute the global namespace with all of the C symbols and possibly lead to confusion if someone is trying to use the C++ library and not touch the C stuff. On the other hand, it is quite nice to be able to reuse things like enums. I guess we can always refine later, just something to think about...

yesterday I did see a post from someone suggesting that you could do something like `namespace detail { #include <foxglove-c/foxglove-c.h> }` to keep the C defs out of the global namespace ... but I'm not sure how kosher that is

Reply:
The namespace trick seems like it works, we can go with that for now